Api additions again (previously KoTM/FRR#100 because github is stupid :> and KoTM/FRR#107 because :>)

lenrik1589 opened the pull request:
1. Adds 3 events:
    - `PROJECTILE_RENDER`
    - `TRIDENT_RENDER`
    - `THROWN_ITEM_RENDER`
2. Adds 2 parent types
    - `TRIDENT`
    - `THROWN_ITEM`
3. Removes unnecessary "isLoaded" check every time `isModLoaded` is executed
4. Adds ability to un-set Action textures by
5. Adds extra data to `client.getDate()`
    - `timestamp`: time in millis since epoch
    - `time`: fully formatted time string
6. Adds ability to give pages mouse events and two new events:
    - `middleClick` fires for middle click mouse button
    -  `click` fires for every mouse button
7. Renames ActionWheelAPI.execute to ActionWheel.mouseClicked to be inline with mouseScrolled used for scrolling, this affects a likely small amount of scripts

Commit:
Api additions

Commit:
Projectiles

Commit:
Minor spelling mistake *dies*

Commit:
added client.getModMetadata
consistent across quilt and fabic:
- id
- name
- description
- values
- version
- icon
- (maybe more, lazy to compare tables)
forgor mixins :woozy_face:

Covkie commented:
Update this pull request to work with the new multisource repo structure.

lenrik1589 commented:
@UnlikePaladin, could you please give pointers as to how it actually works now?

UnlikePaladin commented:
Sure, essentially you can't use or reference mod loader specific classes in the Common package, instead you point to a method of your creation that's annotated with @ExpectPlatform, then you create the impl (if you're using IDEA hovering over the method should provide a shortcut) for each mod loader.

UnlikePaladin commented:
For example in utils we have the PlatformUtils class which contains a isModLoaded method and other methods that reference mod loader classes directly, I also see you're doing some stuff with Mod metadata, for that there's the FiguraMetadata class which is meant to be a container/wrapper for the loader mod metadata, it's currently a bit basic but more methods can be added as needed. Creating and adding new Mixins or LuaJ stuff remains the same

Commit:
fix mod metadata
fix wrong mixin
todo @UnlikePaladin, please fill in PlatformUtilsImpl::getModMetadata for forger

UnlikePaladin commented on fabric/src/main/java/org/figuramc/figura/utils/fabric/PlatformUtilsImpl.java:
I think this is redundant, we already have the FiguraMetadata Class, it just needs expanding with this info

lenrik1589 replied:
resolved by your commit, i guess

UnlikePaladin committed:
Filled in forge metadata and moved everything to the ModMetadataContainer class

Commit:
more input:
add docs for page functions
add middleClick and click
first is called for middle click
second for any mouse button
add ability to unset action texture

lenrik1589 commented:
updated description to reflect the state of things this PR adds/changes

UnlikePaladin committed:
Merge branch '1.20' into feat/new-apis
